.github/workflows/runtests3.py

```diff
@@ -0,0 +1,120 @@
+    
+    # Enable post-update hook
+    post_update_hook = os.path.join(bare_repo_path, 'hooks', 'post-update')
+    with open(post_update_hook, 'w') as f:
+        f.write('#!/bin/sh\n')
+        f.write('git update-server-info\n')
+    os.chmod(post_update_hook, 0o755)
+    subprocess.run(['git', '-C', bare_repo_path, 'update-server-info'], check=True)
+    
+    # Generate SSL certificate
+    keyfile = os.path.join(server_dir, 'key.pem')
+    certfile = os.path.join(server_dir, 'cert.pem')
+    subprocess.run([
+        'openssl', 'req', '-x509', '-newkey', 'rsa:4096',
+        '-keyout', keyfile, '-out', certfile, '-days', '365', '-nodes',
+        '-subj', '/CN=localhost'
+    ], check=True)
+    
+    # Custom Git HTTP handler
+    class GitHTTPRequestHandler(SimpleHTTPRequestHandler):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, directory=bare_repo_path, **kwargs)
+        
+        def do_GET(self):
+            # Handle Git's smart HTTP protocol
+            if self.path.startswith('/repo.git/'):
+                self.path = self.path[len('/repo.git'):]
+            elif self.path == '/repo.git':
+                self.path = '/'
+            return super().do_GET()
+        
+        def end_headers(self):
+            # Add headers required for Git HTTP
+            self.send_header('Content-Type', 'application/x-git-upload-pack-result')
+            super().end_headers()
+
+    # Start HTTPS server
+    port = 8000
+    server = HTTPServer(('localhost', port), GitHTTPRequestHandler)
+    context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+    context.load_cert_chain(certfile=certfile, keyfile=keyfile)
+    server.socket = context.wrap_socket(server.socket, server_side=True)
+    server_thread = threading.Thread(target=server.serve_forever)
+    server_thread.daemon = True
+    server_thread.start()
+    
+    try:
+        TEST_REPO_URL_HTTPS = f"gittuf::https://localhost:{port}/repo.git"
+        TEST_REMOTE_NAME = "origin"
+        GITTUF_TEST_BRANCH = "test-branch"
+
+        print(f"Testing clone using {TEST_REPO_URL_HTTPS}...")
+        run_command(f"git -c http.sslVerify=false clone {TEST_REPO_URL_HTTPS} {TEST_CLONE_DIR}")
+        os.chdir(TEST_CLONE_DIR)
+
+        # Rest of test sequence
+        print("Testing remote configuration...")
+        run_command(f"git remote set-url {TEST_REMOTE_NAME} {TEST_REPO_URL_HTTPS}")
+        
+        print("Testing fetch...")
+        run_command(f"git -c http.sslVerify=false fetch {TEST_REMOTE_NAME}")
+        
+        print("Testing push...")
+        run_command(f"git checkout -b {GITTUF_TEST_BRANCH}")
+        with open("testfile.txt", "w") as f:
+            f.write("test content")
+        run_command("git add testfile.txt")
+        run_command(f"git commit -m 'Add test file'")
+        run_command(f"git push -u {TEST_REMOTE_NAME} {GITTUF_TEST_BRANCH}", 128)
```

Comment:
Likely needs the `sslVerify` option here as well.

```suggestion
        run_command(f"git push -c http.sslVerify=false -u {TEST_REMOTE_NAME} {GITTUF_TEST_BRANCH}", 128)
```